### ISSUE TITLE
[$250] Screen Reader: Many Pages: Focus does not move to opened menu layout (mWeb)

### ISSUE THREAD

**Issue by nbhardwaj-del** [Expensify/App#74858]
## Action Performed
Prerequisite:
- the user is logged in

1. Using TalkBack + Android + Chrome or VoiceOver + iOS + Safari, open the page https://new.expensify.com/settings/about
2. Swipe to navigate to the 'App download links', double tap to activate
3. Using swipe gesture, attempt to navigate through the newly opened menu
4. Observe the focus behavior

Other Occurrences:
1. On Settings - About - Keyboard Shortcuts
2. On Settings - Save the world - I know a teacher
3. On Settings - Save the world - I am a teacher
4. On Settings - Save the world - Intro your school principal
5. On Settings - Preferences - Language
6. On Settings - Preferences - Priority mode
7. On Settings - Security
8. On Settings - Security - Validate your account
9. On Settings - Security - Close account
10. On Settings - Security - Two-factor authentication
11. On Settings - Profile - Display name
12. On Settings - Profile - Contact methods
13. On Settings - Profile - Pronouns
14. On Settings - Profile - Share Code
15. On Settings - Profile - Legal Name
16. On Settings - Profile - DOB
17. On Settings - Profile - Phone number
18. On Settings - Profile - Address
19. On Settings - Profile - Country
20. On Workspaces - Duplicate Workspaces
21. On Workspaces - Delete Workspace
22. On Workspaces - Overview - Workspace Name
23. On Workspaces - Overview - Expensify Policy
24. On Workspace - Categories - Add Category
25. On Workspace - Categories - Settings
26. On Workspace - Workflows - Edit Approval Workflow
27. On Workspace - Workflows - Expenses From
28: On Workspace - Workflows - Approver
29. On Workspace - Rules - Cash Expense Default
30. On Workspaces - Distance Rates - Rate Details
31. On Workspaces - Expensify Card - Add bank account
32. On Workspaces - Expensify Card - Bank info
33. On Workspaces - Expensify Card - Confirm currency and country
34. On Workspace - Company Card - Add Cards
35. On Workspace - Create Workspace - Confirm Workspace
36. On Workspace - Create Workspace - Invite new members
37. On Workspace - Create Workspace - Default Currency
38. On Create Report - Restricted
39. On Create Report - Add payment card
40. On Create Report - Change payment currency
41. On Track Distance
42. On Track Distance - Choose Recipient
43. On Send Invoice
44. On Wallet - Add bank account
45. On Create Expense flow
46. On Paid Expense details flow
47. On Reports flow
48. On Chat flow

## Expected Result
When users activate the 'App download links' button, focus will be contained within the newly opened menu until the user manually closes it

## Actual Result
When users activate the 'App download links' button and use swipe gesture to move through the menu, focus is not contained in the menu, focus moves through the screen behind the menu.

## Area issue was found
Settings-About

## Failed Checkpoint
WCAG 2.4.3 Focus Order
https://www.w3.org/WAI/WCAG22/Understanding/focus-order

## User Impact
People with mobility impairments who must rely on keyboard access for operating a page benefit from a logical, usable focus order. People with vision impairments or disabilities that make reading difficult can become disoriented when tabbing takes focus someplace unexpected or when they cannot easily find the content surrounding an interactive element. Only a small portion of the page may be visible to an individual using a screen magnifier at a high level of magnification. Such a user may interpret a field in the wrong context if the focus order is not logical.

## Suggested Resolution
Ensure focus is moved to the opened layout and trapped inside it until the user closes it.
https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/

## Environment(s)
Motorola RAZR 2023 - Android 14 , Cricket - Chrome

## Screenshots/Videos
https://github.com/user-attachments/assets/35d12a4c-741c-4ec5-ab7d-4cb7fd44b184

<img width="1080" height="2640" alt="Image" src="https://github.com/user-attachments/assets/2597e8a1-1335-413c-bc01-7fd5d340e8c3" />

**Version Number:** NA
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36605/community-issues/6996177
**Issue reported by:** NA
**Slack conversation:** NA

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~022009685628551235822</li>
        <li>Upwork Job ID: 2009685628551235822</li>
        <li>Last Price Increase: 2026-01-09</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @suneox</details>

**Comment by suneox** [comment]
Thank for proposals. this issue has many pages I need more time to verify everything. I’ll finish the review around tomorrow.

**Comment by suneox** [comment]
> @suneox any luck?

I’ll prioritize reviewing this one today.

**Comment by suneox** [comment]
@TaduJR Your proposal doesn’t work as expected. Could you please take another look at this?

https://github.com/user-attachments/assets/ee417d70-84f9-4413-b3a6-d2f0595a1e0a

**Comment by marufsharifi** [comment]
@suneox, thanks for the review. I am checking.

**Comment by marufsharifi** [comment]
@suneox, thanks for the review. This is the [test branch](https://github.com/marufsharifi/App/tree/fix/a11y-focus-opened-menu-mweb). 

It auto focus on content when open via voiceover: 


https://github.com/user-attachments/assets/132e91c0-e0d9-4bfa-9255-4b7f93252bce



> 

@suneox, please give me your feedback. thanks.

**Comment by mallenexpensify** [comment]
> The need https://github.com/Expensify/App/pull/83822 can not resolve issue https://github.com/Expensify/App/issues/83313 we're still waiting for new solution

@rushatgabhane 👀 plz

**Comment by suneox** [comment]
> umm, im not sure what can we do here. [@suneox](https://github.com/suneox) can you suggest a fix for that specific issue?

The remaining issue has been resolved in the PR.
